### ISSUE TITLE
The language server crashes on specific code (C-style aggregate initialization in C++ code) with the French language pack

### ISSUE THREAD

**Issue by alexandrebeaudet** [microsoft/vscode-cpptools#10374]
### Environment

- OS and Version: macOS Ventura 13.1 (22C65)
- VS Code Version:
     tried on 1.75.0-insiders 12515569ae440cf7ccc02d4649a25f2c77909557
     but also fails on "non-insider" version (1.74.3 (Universal) 97dec172d3256f8ca4bfb2143f3f76b503ca0534)
- C/C++ Extension Version: v1.13.9
- If using SSH remote, specify OS of remote machine: N/A

### Bug Summary and Steps to Reproduce

Bug Summary:

The language server crashes and never gets back up when writing specific lines of code that use aggregate initialization and a destructor method (example in additional context).
 
Steps to reproduce:
1. Create a folder
2. Open folder in vscode
3. Create a C++ file (`main.cc`)
4. Write down specific code
5. The language server crashes (see attached screenshot in additional context)

Expected behavior:
The language server doesn't explode

### Configuration and Logs

```shell
No configuration
Log diagnostics unavailable
```

### Other Extensions

None

### Additional context

# Minimal code to test crash :

```cpp
struct Other
{
    ~Other() {} // If this line is commented, works fine
};

struct Internal {
    Other object;
};

struct MyStruct
{
    Internal internal;
};

int main(int argc, char** argv)
{
    MyStruct mystruct {
        // Assigning one step at a time -> ok
        // .internal = {
        //     .object = Other()
        // }

        // Assigning the internal.object directly -> crash
        .internal.object = Other()
    };

    return 0;
}
```

# Screenshot

<img width="819" alt="Capture d’écran 2023-01-11 à 23 09 03" src="https://user-images.githubusercontent.com/91879198/211929961-7b285497-dcba-44e2-9a89-1bbf5fae8612.png">

# Note

My OS is in french, if that might matter. I didn't install VSCode translations for french though (or language server I don't know the name)

Thank you for your time

**Comment by michelleangela** [comment]
@alexandrebeaudet 
I was not able to reproduce the crash an Intel nor Silicon mac.

Could you attach the debugger to the cpptools process and then reproduce the crash to capture a callstack of the crash?
Instructions to attach the debugger at https://github.com/microsoft/vscode-cpptools/wiki/Attaching-debugger-to-cpptools-or-cpptools%E2%80%90srv

**Comment by alexandrebeaudet** [comment]
Ah crap
I tried to capture a callstack but apparently there's another bug on arm mac when attaching to cpptools-srv :(
I'm having the issue on fresh vscode install (I uninstalled the app and cleaned-up all application support files)
It wasn't happening like a month ago I think.
Do you think the installed tools (clang, xcode, ...) could have an influence on the issue ?

**Comment by michelleangela** [comment]
For this crash, the `cpptools` process is the one crashing. It could also be that the `cpptools-srv` crashes as well, but `cpptools-srv` gets created by `cpptools`. There is only one `cpptools` per instance of VS Code, and one or more `cpptools-srv` per `cpptools` process.

Is there also an issue with attaching to `cpptools` (or language server) process? Other installed tools shouldn't cause the process to crash. The language server only queries the clang compiler or other specified compilers for IntelliSense configuration setup. If the language server crashes while querying a compiler, then it should show up in the crash callstack.

Does the process also crash for other code samples or only that particular one? Could you also try installing an older version of the extension such as 1.12.* version and see if the process does not crash with the initial sample code?

**Comment by alexandrebeaudet** [comment]
I'll try installing an older version, I'm pretty sure I was already writing code like that before, and have been using vscode+cpptools for some time now (like 2+ years I think).
I'm not sure what are all the cases where the extension crashes, but it seems very specific to these lines of code. I manage to write almost all the code I want if I take care of *not* writing stuff like `MyStruct smthg = { .outer.inner = <...> };`

**Comment by sean-mcmanus** [comment]
@alexandrebeaudet Your repro code doesn't compile? What are the compiler args you're using to get it to compile (and/or crash our IntelliSense parser)?

**Comment by sean-mcmanus** [comment]
@alexandrebeaudet I reproed the crash after installing the French language pack, so it appears to be trigged by that.

**Comment by sean-mcmanus** [comment]
FYI, without the crash, IntelliSense gives the error: "chained designators are not permitted for a class type with a nontrivial destructor" -- so a workaround may be to fix that error via whatever compiler args you're using to get it to compile.

**Comment by sean-mcmanus** [comment]
It's fixed with 1.14.1 (pre-release). It gives `"les désignateurs chaînés ne sont pas autorisés pour un type classe avec un destructeur non trivial"`. The crash may have been caused by a missing translation for the new error message.
